Integrations/EDL/EDL.py

```diff
@@ -0,0 +1,371 @@
+    Get the ioc list to return in the edl
+    """
+    params = demisto.params()
+    out_format = params.get('format')
+    on_demand = params.get('on_demand')
+    limit = parse_integer(params.get('edl_size'), EDL_LIMIT_ERR_MSG)
+    # on_demand ignores cache
+    if on_demand:
+        values = get_ioc_values_list_from_context(out_format)
+    else:
+        last_run = demisto.getLastRun().get('last_run')
+        indicator_query = demisto.params().get('indicators_query')
+        if last_run:
+            cache_refresh_rate = demisto.params().get('cache_refresh_rate')
+            cache_time, _ = parse_date_range(cache_refresh_rate, to_timestamp=True)
+            if last_run <= cache_time:
+                values = refresh_edl_context(indicator_query, out_format, limit=limit)
+            else:
+                values = get_ioc_values_list_from_context(out_format)
+        else:
+            values = refresh_edl_context(indicator_query, out_format, limit=limit)
+    return values
+
+
+def get_ioc_values_list_from_context(out_format: str) -> list:
+    """
+    Extracts output values from cache
+    """
+    cache_dict = demisto.getIntegrationContext()
+    values = create_csv_out_list(cache_dict) if out_format == FORMAT_CSV else list(cache_dict.values())
+    return values
+
+
+def parse_integer(int_to_parse: Any, err_msg: str) -> int:
+    """
+    Tries to parse an integer, and if fails will throw DemistoException with given err_msg
+    """
+    try:
+        res = int(int_to_parse)
+    except (TypeError, ValueError):
+        raise DemistoException(err_msg)
+    return res
+
+
+''' ROUTE FUNCTIONS '''
+
+
+@APP.route('/', methods=['GET'])
+def route_edl_values() -> Response:
+    """
+    Main handler for values saved in the integration context
+    """
+    params = demisto.params()
+    out_format = params.get('format', 'text')
+    mimetype = 'application/json' if out_format == FORMAT_JSON else 'text/plain'
+    values = list_to_str(get_edl_ioc_values_list(), '\n')
+    return Response(values, status=200, mimetype=mimetype)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(args, params):
+    """
+    Validates:
+        1.
+    """
+    get_params_port(params)
+    cache_refresh_rate = params.get('cache_refresh_rate', '')
+    if cache_refresh_rate:
+        # validate cache_refresh_rate value
+        range_split = cache_refresh_rate.split(' ')
+        if len(range_split) != 2:
+            raise ValueError('Refresh Rate must be "number date_range_unit", examples: (2 hours, 4 minutes,'
+                             '6 months, 1 day, etc.)')
+        parse_integer(range_split[0], 'Refresh Rate number is invalid. Must be a valid integer.')
+        if not range_split[1] in ['minute', 'minutes', 'hour', 'hours', 'day', 'days', 'month', 'months', 'year',
+                                  'years']:
+            raise ValueError(
+                'Refresh Rate time unit is invalid. Must be minutes, hours, days, months or years')
```

Comment:
Invalid time unit for Refresh Rate. Must be minutes, hours, days, months, or years.